src/aiida_common_workflows/workflows/em.py

```diff
@@ -0,0 +1,134 @@
+
+
+def validate_sub_process_class(value, _):
+    """Validate the sub process class."""
+    try:
+        process_class = WorkflowFactory(value)
+    except exceptions.EntryPointError:
+        return f'`{value}` is not a valid or registered workflow entry point.'
+
+    if not inspect.isclass(process_class) or not issubclass(process_class, CommonRelaxWorkChain):
+        return f'`{value}` is not a subclass of the `CommonRelaxWorkChain` common workflow.'
+
+
+def validate_total_magnetizations(value, _):
+    """Validate the `fixed_total_magnetizations` input."""
+    if value and len(value) < 3:
+        return 'need at least 3 total magnetizations.'
+    if not all(isinstance(m, (int, float)) for m in value):
+        return 'all total magnetizations must be numbers (int or float).'
+
+
+def validate_relax_type(value, _):
+    """Validate the `generator_inputs.relax_type` input."""
+    if value is not None and isinstance(value, str):
+        value = RelaxType(value)
+
+    if value not in [RelaxType.NONE, RelaxType.POSITIONS, RelaxType.SHAPE, RelaxType.POSITIONS_SHAPE]:
+        return '`generator_inputs.relax_type`. Equation of state and relaxation with variable volume not compatible.'
+
+
+class EnergyMagnetizationWorkChain(WorkChain):
+    """Workflow to compute the energy vs magnetization curve for a given crystal structure."""
+
+    @classmethod
+    def define(cls, spec):
+        # yapf: disable
+        super().define(spec)
+        spec.input('structure', valid_type=orm.StructureData, help='The structure at equilibrium volume.')
+        spec.input('fixed_total_magnetizations', valid_type=orm.List, required=True,
+            validator=validate_total_magnetizations, serializer=orm.to_aiida_type,
+            help='The list of fixed total magnetizations to be calculated for the structure.')
+        spec.input_namespace('generator_inputs',
+            help='The inputs that will be passed to the input generator of the specified `sub_process`.')
+        spec.input('generator_inputs.engines', valid_type=dict, non_db=True)
+        spec.input('generator_inputs.protocol', valid_type=str, non_db=True,
+            help='The protocol to use when determining the workchain inputs.')
+        spec.input('generator_inputs.relax_type',
+             valid_type=(RelaxType, str), non_db=True, validator=validate_relax_type,
+            help='The type of relaxation to perform.')
+        spec.input('generator_inputs.spin_type', valid_type=(SpinType, str), required=False, non_db=True,
+            help='The type of spin for the calculation.')
+        spec.input('generator_inputs.electronic_type', valid_type=(ElectronicType, str), required=False, non_db=True,
+            help='The type of electronics (insulator/metal) for the calculation.')
+        spec.input(
+            'generator_inputs.fixed_total_cell_magnetization', valid_type=(list, tuple),
```

Comment:
I believe this input should not be exposed, it is set according to the `fixed_total_magnetizations`. Moreover it is of the wrong type, it should be float, no? Or maybe you wanted to expose the `magnetization_per_site`?

Reply:
You're totally right. Just implemented it while mimicking the EoS workflow, but indeed, doesn't make sense here as it is set via `fixed_total_magnetizations`